auto_nag/bzcleaner.py

```diff
@@ -581,37 +582,71 @@ def autofix(self, bugs):
                 if mrg:
                     new_changes[bugid] = mrg
 
-        if self.dryrun or self.test_mode:
-            for bugid, ch in new_changes.items():
-                logger.info(
-                    "The bugs: {}\n will be autofixed with:\n{}".format(bugid, ch)
-                )
-        else:
-            extra = self.get_db_extra()
-            max_retries = utils.get_config("common", "bugzilla_max_retries", 3)
-            for bugid, ch in new_changes.items():
-                added = False
-                for _ in range(max_retries):
-                    failures = bugzilla_cls([str(bugid)]).put(ch)
-                    if failures:
-                        time.sleep(1)
-                    else:
-                        added = True
-                        db.BugChange.add(self.name(), bugid, extra=extra.get(bugid, ""))
-                        break
-                if not added:
-                    self.failure_callback(bugid)
-                    logger.error(
-                        "{}: Cannot put data for bug {} (change => {}).".format(
-                            self.name(), bugid, ch
-                        )
-                    )
+        if not self.apply_autofix:
+            self.autofix_changes = new_changes
+            return bugs
+
+        extra = self.get_db_extra()
+        self.apply_changes_on_bugzilla(
+            self.name(),
+            new_changes,
+            self.no_bugmail,
+            self.dryrun or self.test_mode,
+            extra,
+        )
 
         return bugs
 
-    def failure_callback(self, bugid):
-        """Called on Bugzilla.put failures"""
-        return
+    @staticmethod
+    def apply_changes_on_bugzilla(
+        tool_name: str,
+        new_changes: Dict[str, dict],
+        no_bugmail: bool = False,
+        is_dryrun: bool = True,
+        db_extra: Dict[str, str] = None,
+    ) -> None:
+        """Apply changes on Bugzilla
+
+        Args:
+            tool_name: the name of the tool that are performing the changes.
+            new_changes: the changes that will be performed. The dictatory key
+                should be the bug ID.
+            no_bugmail: If True, an account that doesn't trigger bugmail will be
+                used to apply the changes.
+            is_dryrun: If True, no changes will be applied. Instead, the
+                proposed changes will be logged.
+            db_extra: extra data to be passed to the DB. The dictatory key
```

Comment:
typo

Reply:
Fixed in 762bc5f791659ecc302676ffe54e9d575e0b5902